client/src/components/History/CurrentHistory/HistoryCounter.vue

```diff
@@ -1,67 +1,96 @@
+            </HistoryTargetPreferredObjectStorePopover>
+            <b-button-group>
+                <b-button
+                    v-b-tooltip.hover
+                    title="Show active"
+                    variant="link"
+                    size="sm"
+                    class="rounded-0 text-decoration-none"
+                    @click="setFilter('')">
+                    <span class="fa fa-map-marker" />
+                    <span>{{ numItemsActive }}</span>
+                </b-button>
+                <b-button
+                    v-if="numItemsDeleted"
+                    v-b-tooltip.hover
+                    title="Show deleted"
+                    variant="link"
+                    size="sm"
+                    class="rounded-0 text-decoration-none"
+                    @click="setFilter('deleted:true')">
+                    <icon icon="trash" />
+                    <span>{{ numItemsDeleted }}</span>
+                </b-button>
+                <b-button
+                    v-if="numItemsHidden"
+                    v-b-tooltip.hover
+                    title="Show hidden"
+                    variant="link"
+                    size="sm"
+                    class="rounded-0 text-decoration-none"
+                    @click="setFilter('visible:false')">
+                    <icon icon="eye-slash" />
+                    <span>{{ numItemsHidden }}</span>
+                </b-button>
+                <b-button
+                    v-b-tooltip.hover
+                    :title="reloadButtonTitle"
+                    :variant="reloadButtonVariant"
+                    size="sm"
+                    class="rounded-0 text-decoration-none"
+                    @click="reloadContents()">
+                    <span :class="reloadButtonCls" />
+                </b-button>
+            </b-button-group>
+            <b-modal
+                v-model="showPreferredObjectStoreModal"
+                title="History Preferred Object Store"
+                modal-class="history-preferred-object-store-modal"
+                title-tag="h3"
                 size="sm"
-                class="rounded-0 text-decoration-none"
-                @click="setFilter('visible:false')">
-                <icon icon="eye-slash" />
-                <span>{{ numItemsHidden }}</span>
-            </b-button>
-            <b-button
-                v-b-tooltip.hover
-                :title="reloadButtonTitle"
-                :variant="reloadButtonVariant"
-                size="sm"
-                class="rounded-0 text-decoration-none"
-                @click="reloadContents()">
-                <span :class="reloadButtonCls" />
-            </b-button>
-        </b-button-group>
-    </div>
+                hide-footer>
+                <HistorySelectPreferredObjectStore
+                    :user-preferred-object-store-id="user.preferred_object_store_id"
+                    :history="history"
+                    :root="root"
```

Comment:
This project is incredible. I only have minor comments. We should not parse `root` manually as prop, instead url executing functions should use `safePath` or `prependPath` from `utils/redirect`. Other than that the component and variable names are uniquely long for the UI, which I guess is fine within the newly created components but here I suggest to not use more than 3 words to uniquely identify them and plug them into the history modules.

Reply:
I made these changes in https://github.com/galaxyproject/galaxy/pull/14073/commits/2cdcf156b867aec89d0d1863e4b76585dcaf37d3 - thanks for the tips about root, much cleaner. I prefer the longer component names but I see the other side and I'm find continuing to shorten more but I did start with the history module as requested.